sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/cache/XdsDataCache.java

```diff
@@ -172,31 +210,139 @@ public static void removeRequestObserver(String serviceName) {
     /**
      * update the mapping between service and cluster
      *
-     * @param mapping the mapping between service and cluster
+     * @param clusterMap the map between service and cluster
      */
-    public static void updateServiceNameMapping(Map<String, Set<String>> mapping) {
-        if (mapping == null) {
-            serviceNameMapping = new HashMap<>();
+    public static void updateServiceClusterMap(Map<String, XdsServiceCluster> clusterMap) {
+        if (clusterMap == null) {
+            serviceClusterMap = new HashMap<>();
+            return;
         }
-        serviceNameMapping = mapping;
+        serviceClusterMap = clusterMap;
     }
 
     /**
      * get cluster set for service
      *
-     * @param serviceName
+     * @param serviceName service name
      * @return cluster set for service
      */
     public static Set<String> getClustersByServiceName(String serviceName) {
-        return serviceNameMapping.getOrDefault(serviceName, Collections.EMPTY_SET);
+        XdsServiceCluster xdsServiceCluster = serviceClusterMap.get(serviceName);
+        if (xdsServiceCluster == null) {
+            return Collections.EMPTY_SET;
+        }
+        return xdsServiceCluster.getClusterResources();
+    }
+
+    /**
+     * get serviceClusterMap
+     *
+     * @return serviceClusterMap
+     */
+    public static Map<String, XdsServiceCluster> getServiceClusterMap() {
+        return serviceClusterMap;
+    }
+
+    /**
+     * update HttpConnectionManager
+     *
+     * @param hcms httpConnectionManager list
+     */
+    public static void updateHttpConnectionManagers(List<XdsHttpConnectionManager> hcms) {
+        if (httpConnectionManagers == null) {
+            httpConnectionManagers = new ArrayList<>();
+            return;
+        }
+        httpConnectionManagers = hcms;
+    }
```

Comment:
httpConnectionManagers  is ArrayList. It may cause concurrency issues when update it

Reply:
Updated the hcm list and replaced it with the new hcm list.